### PR TITLE
Game mode hierarchy

### PR DESCRIPTION
**Pull request by JaspervanTilburg**

### CODE REVIEW

File: src/main/java/doob/game/DuelGame.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+				Player p = players.get(i);
+				int lives = p.getLives();
+				int score = p.getScore();
+				level.getPlayers().get(i).setLives(lives);
+				level.getPlayers().get(i).setScore(score);
+			}
+		}
+		readOptions();
+	}
+
```
Comment (cbilstra) [inline src/main/java/doob/game/DuelGame.java]: Do you need javadoc at @override methods?

File: src/main/java/doob/game/GameMode.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	 * handle how a new level is initialized.
+	 */
+	void newLevel();
+	
+	/**
+	 * Handle how to load the highscores.
+	 */
+	void loadHighscores();
+
+}
```
Comment (cbilstra) [inline src/main/java/doob/game/GameMode.java]: I think it is for initialize and updateLives not necessary to  be together. Maybe change this interface to abstract methods?

File: src/main/java/doob/game/DuelGame.java (hunk 1 above)
Comment (jorisoudejans) [inline src/main/java/doob/game/DuelGame.java]: Never hurts, right?

File: src/main/java/doob/game/GameMode.java (hunk 2 above)
Comment (jorisoudejans) [inline src/main/java/doob/game/GameMode.java]: Yeah I agree, just move the initialize and updateLives to the abstract Game class and you get the same effect as now, without needing to have another interface.

File: src/main/java/doob/game/DuelGame.java (hunk 1 above)
Comment (hlycklama) [inline src/main/java/doob/game/DuelGame.java]: Yes you do.

**Comment by jorisoudejans** [comment]
You should test all new Game classes. An abstract test class for game and then extensions of that test class for each game.

**Pull request merged**